### PR TITLE
user creation with username containing capital letters

### PR DESCRIPTION
**Pull request by haribhandari07**
## Description

feature test for user creation using capital letters in username are added
## Related Issue

- Fixes <issue_link>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/features/apiProvisioning-v1/addUser.feature [hunk 1]
```diff
@@ -66,3 +66,17 @@ Feature: add user
     And the HTTP status code should be "200"
     And user "brand-new-user" should exist
     And user "brand-new-user" should be able to access a skeleton file
+
+  Scenario: admin creates a user with username that contains capital letters
+    Given user "brand-new-user" has been deleted
+    When the administrator sends a user creation request for user "BRAND-NEW-USER" password "<password>" using the provisioning API
+    Then the OCS status code should be "100"
+    And the HTTP status code should be "200"
+    And user "brand-new-user" should exist
```
Comment (individual-it) [inline tests/acceptance/features/apiProvisioning-v1/addUser.feature]: please test also that `"BRAND-NEW-USER"` does not exist

**Review by individual-it — CHANGES_REQUESTED** [review]
please add the same tests to v2 API

File: tests/acceptance/features/apiProvisioning-v1/addUser.feature [hunk 2]
```diff
@@ -66,3 +66,17 @@ Feature: add user
...
+
+  Scenario: admin tries to create an existing user but with username containing capital letters
+    Given user "brand-new-user" has been created with default attributes
+    When the administrator sends a user creation request for user "BRAND-NEW-USER" password "%alt1%" using the provisioning API
+    Then the OCS status code should be "102"
+    And the HTTP status code should be "200"
+    And the API should not return any data
```
Comment (phil-davis) [inline tests/acceptance/features/apiProvisioning-v1/addUser.feature]: Could also check here that nothing strange happened - that the original user is still there...
```
And user "brand-new-user" should exist
And user "BRAND-NEW-USER" should not exist
```

**Review by skshetry — COMMENTED** [review]
:+1:

**Comment by individual-it** [comment]
@ownclouders rebase

File: tests/acceptance/features/apiProvisioning-v1/addUser.feature (hunk 2 above)
Comment (individual-it) [reply tests/acceptance/features/apiProvisioning-v1/addUser.feature]: it will exist, because the API does not care about how you send the username

**Review by individual-it — APPROVED** [review]
please squash before merging

**Comment by HariBhandari07** [comment]
Backport on #35276

**Pull request merged**